walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,70 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+    final long MINUTES_IN_A_DAY = 1440;
+    final int START_OF_DAY = TimeRange.START_OF_DAY;
+    final int END_OF_DAY = TimeRange.END_OF_DAY;
+    int endOfTimeRange = 0;
+    TimeRange previousTimeRange = TimeRange.fromStartEnd(0,0,true);
+    Collection<TimeRange> unavailableTimeRange = new ArrayList<>();
+    Collection<TimeRange> unavailableNoOverlap = new ArrayList<>();
+    Collection<TimeRange> returnCollection = new ArrayList<>();
+
+    long durationOfRequest = request.getDuration();
+    Collection<String> attendeesOfRequest = request.getAttendees();
+
+    if(attendeesOfRequest.isEmpty()){
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }else if(durationOfRequest > MINUTES_IN_A_DAY){
+      return Arrays.asList();
+    }else{
+      for(String attendee : attendeesOfRequest){
+        for(Event event : events){
+          if(event.getAttendees().contains(attendee)){
+            unavailableTimeRange.add(event.getWhen());
+          }
+        }
+      }
+      for(TimeRange timerange : unavailableTimeRange){
+        if(previousTimeRange.contains(timerange)){
+        }else if(previousTimeRange.overlaps(timerange)){
+          TimeRange newTimeRange = TimeRange.fromStartEnd(previousTimeRange.start(),timerange.end(),false);
+          unavailableNoOverlap.add(newTimeRange);
+          unavailableNoOverlap.remove(previousTimeRange);
+          previousTimeRange = newTimeRange;
+        }else{
+          unavailableNoOverlap.add(timerange);
+          previousTimeRange = timerange;
+        }
+      }
+      for(TimeRange timerange : unavailableNoOverlap){
+        if(endOfTimeRange == 0){
+          TimeRange trval = TimeRange.fromStartEnd(START_OF_DAY,timerange.start(),false);
+          if(trval.duration() != 0 && trval.duration() >= durationOfRequest){
```

Comment:
you don't have to check `trval.duration != 0`, since that will automatically be check in the next condition `trval.duration() >= durationOfRequest`, since durationOfRequest should be greater than 0. If durationOfRequest is  0 or negative, I'd prefer including an edge case check for that at the beginning.